docs/logging.md

```diff
@@ -1,10 +1,65 @@
-# Logging and Performance Debugging
+# Logging and Error reporting
 
-It's good to have the terminal running `run-dev.py` up as you work since error
-messages including tracebacks along with every backend request will be printed
-there.
+Having a good system for logging error reporting is essential to
+making a large project like Zulip successful.  Without reliable error
+reporting, one has to rely solely on bug reports from users in order
+to produce a working product.
 
-The messages will look similar to:
+Our goal as a project is to have 0 known 500 errors on the backend and
+0 known JavaScript exceptions on the frontend.  While there will
+always be new bugs being introduced, that goal is impossible without
```

Comment:
Maybe word this more like "we can get closer to the goal with..."

Reply:
I think the idea is that the goal of 0 is actually achievable -- not to always be at 0, but for the steady state to be at 0, where after a bug is introduced we spot it, fix it, and get back to 0.

s/While/Because/ would make the logic work, I think.